keps/2349-multikueue-external-custom-job-support/README.md

```diff
@@ -0,0 +1,153 @@
+    - [Risks](#risks)
+    - [Mitigations](#mitigations)
+- [Design Details](#design-details)
+  - [Generic Adapter via ConfigMap](#generic-adapter-via-configmap)
+  - [Workload-Only Sync for External Controllers](#workload-only-sync-for-external-controllers)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+ This KEP introduces a mechanism to support MultiKueue for custom job frome external controllers without requiring modifications changes in external job controllers.  This introduces a generic, built-in MultiKueue adapter that is configurable via a configmap.
+
+## Motivation
+
+Many Kueue adopters have specific CustomResources which might not be public.
+Kueue should give a possibility to manage such CustomResources across multiple clusters.
+
+### Goals
+
+Give a possibility for external controllers to manage private CustomResources in a MultiKueue environment.
+
+
+### Non-Goals
+
+## Proposal
+
+MultiKueue controller will be configurable via configmap. Based on this configmap, a generic MultiKueueAdapter will be implemented within Kueue. When processing a workload without supported built-in adapter, generic MultiKueueAdapter will be called based on GroupVersionKind (GVK) defined in configmap. It will use configuration defined to manage lifycycle's on the worker clusters.
+
+### User Stories ( Optional )
+Ability to schedule resources not yet supported by kueue like Tekton.
+
+### Constraints
+
+- Limited Transformation: Generic Adaptor are limited to simple JsonPatch transformation configurable.
+
+### Risks and Mitigations
+
+#### Risks
+
+ Configmaps are unstructured by default. They don't provide validation by default unlike CRD. User could make typo mistakes or give invalid syntax in configmap. Or user could give invalid syntax.
+
+#### Mitigations
+
+Configmap which aren't correct should logged like Invalid GVK format, Invalid JsonPatch ops. We could have cli command to validate adapter logic.
+
+## Design Details
+
+### Generic Adapter via ConfigMap
+
+The `kueue-manager-config` configmap will be expanded to configure the generic adapter.
+
+```
+     # In the main Kueue ConfigMap
+     integrations:
+       externalFrameworks:
+         # OLD FORMAT
+         - "taskrun.tekton.dev"
+         # NEW, EXPANDED FORMAT (for MultiKueue)
+         - group: "pipelinerun.tekton.dev"
+           version: "v1"
+           kind: "PipelineRun"
+           multiKueueAdapter:
+            managedBy: ".spec.managedBy"
+            creationPatches:
+            # Used to define what to patch while creating in worker. // By default, we can assume replace with null for managedBy field.
+                - op: replace
```

Comment:
I believe these fields could be default actually, bringing down the user-facing configuration for most framewrorks to minimum - just the enablement entry. Still, it is valueable to expose the configuration for frameworks which would have the fields in different places.

Reply:
Yes. A default value makes sense.

Reply:
I added point about default values.